### ISSUE TITLE
AnalogServo driver 

### ISSUE THREAD

**Issue by to92me** [bstaletic/eurobot2k17#4]
propose API and implement driver.

**Comment by bstaletic** [comment]
This comes down to just calling `timer_set_oc_output()` as analog servo motors use plain PWM as input.

My idea was to use a desired angle as an input parameter for something like `rotate_analog_servo()` and somehow calculate PWM output compare value from the angle.

We could also have two functions for rotating by absolute and relative degrees.

On a side note, I think it is best to define each servo according to its function in `AnalogServo.h`, like it is already done for USART2 in [`motion.h`](https://github.com/bstaletic/eurobot2k17/blob/master/src/drivers/actuators/motion/motion.h#L8).

Does the driver need a function to retrun the current (or previously set) position?

**Comment by to92me** [comment]
I agree on two functions with absolute and relative degrees.

In my opinion there is no need of returning old position. It could return error or success status but i don't think that hardware errors will implemented sooo void or int for future enhancements ? @bstaletic. 

Or maybe one parameterized driver for all analog servos. Analog servo API first argument will be struct with data of servo. This way parameters of servo will be i same place as logic of executor.

**Comment by darko31** [comment]
I don't think there is need for relative coordinates. One parameterized function sounds good. Function with two arguments, servo output number from 0 to 7, and position in degrees from 0 to 180 scaled to PWM value. There is one problem for selecting right servo output as couple of different timers control different  servo outputs. Boris configured them.

**Comment by bstaletic** [comment]
> In my opinion there is no need of returning old position.

I agree, that was only a suggestion.

> It could return error or success

Do we have any way of determining this? We are talking about simply setting a PWM output compare value.

> one parametrized driver

Haven't thought of that. Sounds like a good idea.

`timer_set_oc_value()` takes timer as the first parameter (i.e. `TIM1`), PWM channel as the second (i.e. `TIM_OC1`) and, of course, value as the final parameter.

Easiest way of implementing this would be a bunch of `if/else` statements. Though, that seems like a hack(ish) approach. Any better ideas?

**Comment by darko31** [comment]
There is no way of knowing if analog servo moved to the right position so there's no need for return success. If/else statements sound good. If you really want to make it a bit prettier, maybe using some kind of defines could be a solution.

**Comment by to92me** [comment]
well I thought something like this:
``` 
Struct ServoAnalog{
     Timer timer,
     Pwm channel, 
     etc parameters.. 
} 
```
and API: 
``` move(ServoAnalog _sa, Int _angle); ```  
and then in Executor you create struct with corresponding data and use AnalogServo API. 
I am used to this kind of API and because of that I am proposing it.  
Another method is to create void pointers in struct, "constructor", "methods" ...

**Comment by darko31** [comment]
Yeah, this style with struct look good.

**Comment by bstaletic** [comment]
Looks good to me as well.

**Comment by bstaletic** [comment]
This has been implemented in #27.
